syntax = "proto3";

import "base/game_base.proto";
option optimize_for = LITE_RUNTIME;

// 创建房间
message BaiRenCreateRoom {
    int32 baseScore = 1; // 底注设定
    int32 inScore = 2; // 入场设定
    int32 leaveScore = 3; // 离场场设定
}

message BaiRenDice {
    repeated int32 dice = 1; // 骰子
}

// 发牌
message BaiRenDealCardAction {
    repeated int32 cards = 1;
    int32 cardSize = 2;
    repeated int32 dealedCards = 3;
}
//询问
message BaiRenRecAsk {
    int32 time = 1; //下注时间
    int32 type = 2; //1.骰子 2.下注
}

// 下注
message BaiRenBetScore {
    int32 score = 1;
    int32 index = 2; //下注方位
    int32 playerId = 3;
}

message BaiRenBetScoreAction {
    repeated BaiRenBetScore betScore = 1;
}

message BaiRenPositions {
    repeated BaiRenBetScore positions = 1;
    int32 shensuanziPositions = 2;
}

//单局结算
message BaiRenPlayerOneSetResult {
    repeated BaiRenSettlePlayerInfo players = 1;
    RecUpdateGameUsers.UserInfo dayingjia = 2;
    int32 bankerWin = 3; //1.通赔 2.通杀
    repeated int32 positionWin = 4; //0.平局 1.赢 2.输
    RecUpdateGameUsers.UserInfo dashujia = 5;
    int32 dayingjiaScore = 6;
    int32 dashujiaScore = 7;

}

message BaiRenSettlePlayerInfo {
    int32 playerId = 1;
    int32 score = 2; // 当局分数
    int32 totalScore = 3;
}

message BaiRenScore {
    repeated int32 score = 1; // 当局分数
}

message BaiRenWatchSize {
    int32 watchSize = 1; // 无座玩家数量
}

message BaiRenTrend {
    message SigleTrend {
        repeated int32 positionWin = 1; //0.平局 1.赢 2.输
    }
    repeated SigleTrend trends = 1;
}

message BankerConfirm {
    RecUpdateGameUsers.UserInfo banker = 1;
    int32 shangzhuangScore = 2;
    int32 bankerTimes = 3;
}

message ShangZhuangList {
    repeated BankerConfirm bankerList = 1;
}